test/test-parse-warning.cpp

```diff
@@ -436,6 +436,33 @@ TEST_F(Test_Parse_Warning, warn_on_xor_operation_used_as_exponentiation) {
   test_parse_and_visit_expression(u8"4 ^ 3"_sv, no_diags);
   test_parse_and_visit_expression(u8"(x+2)^a"_sv, no_diags);
 }
+TEST_F(Test_Parse_Warning, Diag_Explicit_Fallthrough_Comment_In_Switch) {
+  test_parse_and_visit_statement(
+      u8"switch(cond1){case 1:\nfoo()\ncase 2:\nbar() //fallthrough\ndefault:}"_sv,  //
+      u8"                       ^^^ Diag_Explicit_Fallthrough_Comment_In_Switch"_diag);
```

Comment:
Must fix: I think the diagnostic is in the wrong place. As a user, I would expect the diagnostic to be between the statement and the following `case`, not at the beginning of the statement. In other words, the diagnostic should point me to where I should fix the problem. For example, the diagnostic could be just before `case`:
```c++
      u8"switch(cond1){case 1:\nfoo()\ncase 2:\nbar() //fallthrough\ndefault:}"_sv,  //
      u8"                              ` Diag_Explicit_Fallthrough_Comment_In_Switch"_diag);
```

Reply:
You are right. I went with the suggestion of adding `has_leading_comment` 👍 
Also switched the diag to be just before `case` or `default`.